### ISSUE TITLE
Hastus import: some routes missing, and some stops missing from some routes

### ISSUE THREAD

**Issue by sherson** [umts/stop-project#97]
Email excerpts from Alex Forrest <AlexForrest@pvpc.org>

> not all routes are showing up in the route selection dropdown, missing the following:
> 
> * Route 33
> * Route P20E (aka “920”)
> * Route R41
> * Route B48
> * Route X98
> * Nashawannuck Express
> * Palmer Village
> * Ware Shuttle
> 
> Also, a few routes are extraneous:
>  
> * 51
> * Routes beginning with “8” (801 thru 892)

and

> Running into a more serious problem re: the B43, where many westbound stops simply aren’t showing up at all. When I sort by sequence, the westbound trip goes from Haigis Mall directly to the Northampton city line, lists the Northampton stops, and then concludes with the Hadley town line, rather than with Smith College. Consequently, all the westbound Hadley stops are missing—when I search for them by Stop ID instead, I can find them, but they don’t list any routes which serve them (which I assume is why they aren’t showing up in the B43 sequence).
> 
> I’m not sure how many other routes and stops have similar issues, or whether this problem originates in Hastus or just came up during the import. It’s also not an absolute obstacle to stop inventorying—it just means that I’ll need to look up any such stops individually, rather than relying on the lists by route. Because it’s already so late in June, I’ll be starting the inventory today with our intern

I confirmed that [stops_with_routes_v2.exp](https://umass.box.com/s/q8xfvpusdaoei3hpsp60xtjg6reo9meo) is missing those routes, and some stops from some routes. For example, here are all the rows for B43 in that file:

```
B43;EXSAC;West;116;1
B43;EXSAC;West;10001;2
B43;EXSAC;West;9169;3
B43;EXSAC;West;229;4
B43;EXSAC;West;238;5
B43;EXSAC;West;254;6
B43;EXSAC;West;260;7
B43;EXSAC;West;252;8
B43;EXSAC;West;251;9
B43;EW;West;116;1
B43;EW;West;112;2
B43;EW;West;96;3
B43;EW;West;94;4
B43;EW;West;81;5
B43;EW;West;73;6
B43;EW;West;9169;7
B43;EW;West;229;8
B43;EW;West;238;9
B43;EW;West;254;10
B43;EW;West;260;11
B43;EW;West;252;12
B43;EW;West;251;13
B43;ACX;East;251;1
B43;ACX;East;261;2
B43;ACX;East;244;3
B43;ACX;East;230;4
B43;ACX;East;9171;5
B43;ACX;East;73;6
B43;R;East;251;1
B43;R;East;261;2
B43;R;East;244;3
B43;R;East;230;4
B43;R;East;9170;5
B43;R;East;183;6
B43;R;East;176;7
B43;R;East;174;8
B43;R;East;172;9
B43;R;East;168;10
B43;R;East;162;11
B43;R;East;160;12
B43;R;East;146;13
B43;R;East;136;14
B43;R;East;130;15
B43;R;East;10001;16
B43;R;East;128;17
B43;R;East;124;18
B43;R;East;87;19
B43;R;East;78;20
B43;R;East;73;21
B43;R;East;75;22
B43;R;East;82;23
B43;R;East;93;24
B43;R;East;95;25
B43;R;East;113;26
B43;R;East;116;27
B43;R no Ham;East;251;1
B43;R no Ham;East;261;2
B43;R no Ham;East;244;3
B43;R no Ham;East;230;4
B43;R no Ham;East;9170;5
B43;R no Ham;East;183;6
B43;R no Ham;East;176;7
B43;R no Ham;East;174;8
B43;R no Ham;East;172;9
B43;R no Ham;East;168;10
B43;R no Ham;East;162;11
B43;R no Ham;East;160;12
B43;R no Ham;East;146;13
B43;R no Ham;East;136;14
B43;R no Ham;East;130;15
B43;R no Ham;East;10001;16
B43;R no Ham;East;128;17
B43;R no Ham;East;124;18
B43;R no Ham;East;87;19
B43;R no Ham;East;78;20
B43;R no Ham;East;73;21
B43;R no Ham;East;75;22
B43;R no Ham;East;82;23
B43;R no Ham;East;93;24
B43;R no Ham;East;95;25
B43;R no Ham;East;113;26
B43;R no Ham;East;116;27
```

From spot checking, what's there seems to match what's in the app. Using https://bustracker.pvta.com/InfoPoint/, I can see that several stops are on the B43 westbound in Hadley that aren't in the export file, just like Alex said. For example:

144
159
169
171
173
175
178

I exported using 5/6-12 as a date range as per SATCo's instructions. I have not yet had a chance to look in Hastus. @dfaulken is this something you might have time to tackle?

Related note: the newest control files are not at https://github.com/umts/stop-project/tree/master/contrib yet.

**Comment by dfaulken** [comment]
> Related note: the newest control files are not at https://github.com/umts/stop-project/tree/master/contrib yet.

Where are they?

**Comment by sherson** [comment]
I got copies of them from you. I put them on the Hastus server when I ran them.... hang on and I'll look where.

I forgot to mention that my thought was that we should rerun the export and see if we can reproduce the problem. I dunno if there's something wrong with the control file, something wrong with the date range, something else wrong in Hastus, or if I screwed something up when I ran the export.

**Comment by dfaulken** [comment]
The intent of my question was: if they are producing data that we are loading into production, whose state is reflected by the master branch, why are they not on GH, in the master branch?

**Comment by sherson** [comment]
Because you were the last person to modify those files, I thought you were going commit them. 

And you did (11 days ago... I just missed that). Oops. Sorry for the misunderstanding.

**Comment by dfaulken** [comment]
Ah. I thought you meant they had been updated since then. Glad we're on the same page.

**Comment by sherson** [comment]
Right, my bad. 

At first I was all like "Sheesh, why is Dave asking me why his files weren't committed?"

And then I was all like "Oh. Of course he committed them. I'm a dumbass."

**Comment by dfaulken** [comment]
So first question here:

Can we confirm that trips running those routes were scheduled during the date range from which vehicle schedules were pulled for this export?

**Comment by sherson** [comment]
> Can we confirm that trips running those routes were scheduled during the date range from which vehicle schedules were pulled for this export?

We talked on the phone shortly after you posted, but just to update the issue: I checked Hastus and confirmed that at least 33 ran during those dates. It was the finals week schedule, which lacks tripper, but there's no reason the 33 should've been excluded. I haven't had time to rerun the export for other dates to see if we get different results.

I asked Bridgette if she had any ideas, and nothing came to mind.

**Comment by dfaulken** [comment]
Okay. Then I think the next step is to come up with a specific example of a data point that wasn't in the file, and that we can confirm should have been in the file, and to send the details to GIRO for their support.

**Comment by dfaulken** [comment]
@mboneil10 addressed by using GTFS, correct?

**Comment by mboneil10** [comment]
👍 Closed via #101
